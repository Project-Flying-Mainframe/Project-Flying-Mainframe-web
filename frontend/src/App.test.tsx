import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

describe('App Component', () => {
  it('renders Home link', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const linkElement = screen.getByRole('link', { name: 'Home' });
    expect(linkElement).toBeInTheDocument();
  });

  it('renders Catalog link', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const linkElement = screen.getByRole('link', { name: 'Catalog' });
    expect(linkElement).toBeInTheDocument();
  });
});
