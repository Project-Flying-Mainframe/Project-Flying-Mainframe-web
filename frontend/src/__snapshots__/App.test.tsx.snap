// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-container"
  >
    <header
      class="header"
    >
      <div
        class="brand"
      >
        <button>
          ☰
        </button>
        <a
          href="index.html"
        >
          Project Flying Mainframe
        </a>
      </div>
      <div
        class="header-links"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/catalog"
        >
          Catalog
        </a>
        <a
          href="cart.html"
        >
          Cart
        </a>
        <a
          href="signin.html"
        >
          Sign In
        </a>
      </div>
    </header>
    <aside
      class="sidebar"
    >
      <h3>
        Shopping Categories
      </h3>
      <button
        class="sidebar-close-button"
      >
        x
      </button>
      <ul>
        <li>
          <a
            href="index.html"
          >
            Pants
          </a>
        </li>
        <li>
          <a
            href="index.html"
          >
            Shirts
          </a>
        </li>
      </ul>
    </aside>
    <main
      class="main"
    >
      <div>
        <h2>
          Home
        </h2>
      </div>
    </main>
    <footer>
      © 2022 Project Flying Mainframe
    </footer>
  </div>
</DocumentFragment>
`;
